src/db.c

```diff
@@ -1959,8 +2025,8 @@ static struct db_sys_list *_db_rule_gen_32(const struct arch_def *arch,
 	unsigned int iter;
 	struct db_sys_list *s_new;
 	const struct db_api_arg *chain = rule->args;
-	struct db_arg_chain_tree *c_iter = NULL, *c_prev = NULL;
-	bool tf_flag;
+	struct db_arg_chain_tree *prev_nodes[4] = { NULL, };
```

Comment:
Another note to self - the `prev_nodes[]` change will become important in later commits.  This commit is (slightly) more than just a refactoring of code to a separate function.

Reply:
The fixed `4` is very magic number too, which makes me uneasy.  At the very least that needs a comment (why 4?), but a much better approach would be a macro definition *and* and comment with the macro.

Reply:
Okay, I now understand where the `4` comes from, and to be fair we have similar constants in the current `_db_rule_gen_64()` function, but the important difference is that in the current implementation the magic numbers are contained as local variables within the function, here they are part of the function's signature and are thus much more important.

Reply:
I'm hoping that as I get farther into this review we can revisit how this code was extracted, but I'm not really liking this right now.